### ISSUE TITLE
A yoloV3 loss Bug？

### ISSUE THREAD

**Issue by sumenpuyuan** [dmlc/gluon-cv#832]
![image](https://user-images.githubusercontent.com/14249563/60007020-143f3380-96a4-11e9-98ee-907c014cbae1.png)
如上图，
（1）第一行代码感觉没有任何意义
（2）yolo v3 忽略区域也要惩罚置信度，这里没有进行惩罚？
是我代码理解有问题，还是代码书写有些问题？

**Comment by zhreshold** [comment]
Anchors that has IOU in `range(ignore_iou_thresh, pos_iou_thresh)` don't get penalized for objectness score.

**Comment by yuzhms** [comment]
I don't understand why ignored anchor don't calculate loss, can you give a reference of papers or codes in original yolo darkness implementation?
@zhreshold

**Comment by sumenpuyuan** [comment]
是我读论文疏忽的问题，确实忽略区域不惩罚置信度。（我以前认为忽略区域惩罚置信度，但不惩罚边界框回归），下图是yolov3原始论文,绿色的部分可能会对红色部分产生混淆。但事实是论文提到忽略他，即就是既不惩罚，也不奖励。
![image](https://user-images.githubusercontent.com/14249563/60417874-13fae700-9c14-11e9-9419-846b8400ca19.png)
另外，我再keras实现的代码中找到了合理的实现,可见到忽略区域置信度是不进行惩罚的。
```python
# Find ignore mask, iterate over each of batch.
        ignore_mask = tf.TensorArray(K.dtype(y_true[0]), size=1, dynamic_size=True)
        object_mask_bool = K.cast(object_mask, 'bool')
        def loop_body(b, ignore_mask):
            true_box = tf.boolean_mask(y_true[l][b,...,0:4], object_mask_bool[b,...,0])
            iou = box_iou(pred_box[b], true_box)
            best_iou = K.max(iou, axis=-1)
            ignore_mask = ignore_mask.write(b, K.cast(best_iou<ignore_thresh, K.dtype(true_box)))
            return b+1, ignore_mask
        _, ignore_mask = K.control_flow_ops.while_loop(lambda b,*args: b<m, loop_body, [0, ignore_mask])
        ignore_mask = ignore_mask.stack()
        ignore_mask = K.expand_dims(ignore_mask, -1)
confidence_loss = object_mask * K.binary_crossentropy(object_mask, raw_pred[...,4:5], from_logits=True)+ \
            (1-object_mask) * K.binary_crossentropy(object_mask, raw_pred[...,4:5], from_logits=True) * ignore_mask
```
